tests/Unit/Domain/Test_SegmentId.cpp

```diff
@@ -0,0 +1,35 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <catch.hpp>
+
+#include "Domain/SegmentId.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+
+TEST_CASE("Unit.Domain.SegmentId", "[Domain][Unit]") {
+  for (size_t level = 1; level < 5; ++level) {
+    const double segment_length = 2.0 / two_to_the(level);
+    double midpoint = -1.0 + 0.5 * segment_length;
+    for (size_t segment_index = 0; segment_index < two_to_the(level);
+         ++segment_index) {
+      SegmentId id(level, segment_index);
+      CHECK(id.midpoint() == midpoint);
+      midpoint += segment_length;
+      CHECK(id.refinement_level() ==
```

Comment:
All these checks are duplicating the equality operator logic.  How about testing the equality operator and then using it in the later checks?